sdk/core/azure-core/inc/azure/core/context.hpp

```diff
@@ -41,7 +40,35 @@ namespace Azure { namespace Core {
   };
 
   /**
-   * @brief A context is a node within a tree that represents deadlines and key/value pairs.
+   * @brief A context is a node within a unidirectional tree that represents deadlines and key/value
+   * pairs.
+   *
+   * Most Azure Service Client operations accept a Context object. The Context object allows the
+   * caller to cancel the operation if it is taking too long, or to ensure that the operation
+   * completes before a specific deadline. This allows an application to apply timeouts to
+   * operations, or to cancel operations that need to be abandoned.
+   *
+   * After cancelling a Context, all service operations operations which have the cancelled context
+   * as a parent context will be cancelled. Cancellation is indicated by throwing an
+   * Azure::Core::OperationCancelledException from the operation.
+   *
+   * Context objects support the following operations to create new contexts:
+   * - WithDeadline(DateTime): creates a new child context with a deadline.
+   * - WithValue(Key, T): creates a new child context with a key/value pair.
+   * - CreateWithDeadline(DateTime): Static method to create a new context with a deadline.
```

Comment:
Update document, now it's a ctor.

Reply:
Removed text. Not going to add the ctor documentation here, since this is an overview.